shared/naturalcrit/codeEditor/codeEditor.jsx

```diff
@@ -169,9 +146,61 @@ const CodeEditor = createClass({
 		});
 		closeTag.autoCloseCurlyBraces(CodeMirror, this.codeMirror);
 
+		this.setState({
```

Comment:
I'm not sure `widgetUtils` needs to be in the component state, since it looks like it's only set once here when the component is first mounted. State is meant to hold variables that should trigger a re-render of the component when they change, but I don't think widgetUtils ever changes after this first `require`.

Reply:
I did that so the widget utils are defined per-component instance. I know technically I can throw them on the class as a member var but I felt putting it in the state made more sense because I've noticed some odd things with component member vars